### PR TITLE
fix(L-05): remove redundant error from ECDSA smart contract

### LINKED ISSUE #328: [Audit]: L-05 Missing Signature Length Check
### What is the feature you would like to see?

The ECDSA library does not utilize the  `ECDSAInvalidSignatureLength`  error. According to
the Solidity library, the concatenation of `v` ,  `r` , and  `s`  must have a valid signature length of 65
bytes, otherwise, an error is returned. This validation is missing in the  `_recover`  function
within  ecdsa.rs , where the signature length is not checked.
Consider adding this validation check for the correct signature length in the  `_recover`
function. In addition, consider correcting the misplaced comment referencing this error in line
189.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines

### PR DESCRIPTION
**Pull request by bidzyyys**
Resolves #328

### CODE REVIEW

**Review by ggonzalez94 — APPROVED** [review]

File: contracts/src/utils/cryptography/ecdsa.rs [hunk 1]
```diff
@@ -186,7 +177,7 @@ fn encode_calldata(hash: B256, v: u8, r: B256, s: B256) -> Vec<u8> {
 /// and for v in (302): v ∈ {27, 28}.
 ///
 /// Most signatures from current libraries generate a unique signature
-/// with an s-value in the lower half order.ECDSAInvalidSignatureLength
+/// with an s-value in the lower half order.
```
Comment (qalisander) [inline contracts/src/utils/cryptography/ecdsa.rs:180]: isn't here slightly different meaning?

Comment (qalisander) [reply contracts/src/utils/cryptography/ecdsa.rs:180]: I see It wasn't grammatically correct before..

**Review by qalisander — APPROVED** [review]
LGTM

File: contracts/src/utils/cryptography/ecdsa.rs (hunk 1 above)
Comment (bidzyyys) [reply contracts/src/utils/cryptography/ecdsa.rs:180]: `ECDSAInvalidSignatureLength` should not be placed there 😅

**Review by ggonzalez94 — APPROVED** [review]

**Pull request merged**